models/model_choice.py

```diff
@@ -153,19 +215,53 @@ def net(net_params, num_channels, inference=False):
         radius_channel = get_key_def('coordconv_radius_channel', net_params['global'], False)
         scale = get_key_def('coordconv_scale', net_params['global'], 1.0)
         # note: this operation will not attempt to preserve already-loaded model parameters!
-        model = coordconv.swap_coordconv_layers(
-            model, centered=centered, normalized=normalized, noise=noise,
-            radius_channel=radius_channel, scale=scale
-        )
+        model = coordconv.swap_coordconv_layers(model, centered=centered, normalized=normalized, noise=noise,
+                                                radius_channel=radius_channel, scale=scale)
 
     if inference:
         state_dict_path = net_params['inference']['state_dict_path']
         assert Path(net_params['inference']['state_dict_path']).is_file(), f"Could not locate {net_params['inference']['state_dict_path']}"
         checkpoint = load_checkpoint(state_dict_path)
-    elif train_state_dict_path is not None:
-        assert Path(train_state_dict_path).is_file(), f'Could not locate checkpoint at {train_state_dict_path}'
-        checkpoint = load_checkpoint(train_state_dict_path)
+
+        return model, checkpoint, model_name
+
     else:
-        checkpoint = None
 
-    return model, checkpoint, model_name
+        if train_state_dict_path is not None:
+            assert Path(train_state_dict_path).is_file(), f'Could not locate checkpoint at {train_state_dict_path}'
+            checkpoint = load_checkpoint(train_state_dict_path)
+        else:
+            checkpoint = None
+        assert num_devices is not None and num_devices >= 0, "missing mandatory num gpus parameter"
+        # list of GPU devices that are available and unused. If no GPUs, returns empty list
+        lst_device_ids = get_device_ids(num_devices) if torch.cuda.is_available() else []
+        num_devices = len(lst_device_ids) if lst_device_ids else 0
+        device = torch.device(f'cuda:{lst_device_ids[0]}' if torch.cuda.is_available() and lst_device_ids else 'cpu')
+        print(f"Number of cuda devices requested: {net_params['global']['num_gpus']}. Cuda devices available: {lst_device_ids}\n")
+        if num_devices == 1:
+            print(f"Using Cuda device {lst_device_ids[0]}\n")
+        elif num_devices > 1:
```

Comment:
I understand. This is where model is pushed to GPU if available. Conforms docstring in load_checkpoint should be updated.

Reply:
Done